source/s3/s3.go

```diff
@@ -0,0 +1,61 @@
+package s3
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/MiLk/nsscache-go/cache"
+	"github.com/MiLk/nsscache-go/source"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/pkg/errors"
+)
+
+type S3Source struct {
+	prefix string
+	bucket string
+	client s3iface.S3API
+}
+
+func CreateS3Source(client s3iface.S3API, prefix string, bucket string) source.Source {
```

Comment:
Please follow the same pattern as https://github.com/MiLk/nsscache-go/blob/master/source/vault/vault.go#L31
And use functional options instead parameters like you did.